In [1]:
time = []
distance = []
with open('input.txt') as f:
    for line in f:
        if line.startswith("Time"):
            time = [int(n) for n in line[len("Time:"):].strip().split(' ') if n != '']
        if line.startswith("Distance"):
            distance = [int(n) for n in line[len("Distance:"):].strip().split(' ') if n != '']
td = []
for i in range(len(time)):
    td.append({'time':time[i], 'distance':distance[i]})
    
start_speed = 0
speed_increase = 1

total_way = 1
for e in td:
    way = 0
    t = e['time']
    d = e['distance']
    
    for i in range(t):
        dist = (start_speed+speed_increase*i)* (t-i)
        #print("distance:", dist)
        if dist > d:
            way += 1
    total_way *= way
        
print(total_way)

275724


## Part 2

In [1]:
import numpy as np

In [7]:
time = []
distance = []
with open('input.txt') as f:
    for line in f:
        if line.startswith("Time"):
            time = int("".join([(n) for n in line[len("Time:"):].strip().split(' ') if n != '']))
            print(time)
        if line.startswith("Distance"):
            distance = int("".join([(n) for n in line[len("Distance:"):].strip().split(' ') if n != '']))
            print(distance)
    
start_speed = 0
speed_increase = 1

way = 0

for i in range(time):
    dist = (start_speed+speed_increase*i)* (time-i)
    #print(dist)
    #print("distance:", dist)
    if dist > distance:
        way += 1
    else:
        if way > 0:
            break

print(way)

59688274
543102016641022
37286485


first try : 543102016641022 too high : ... it was the input, the code did not finished

second try: 37286485 OK